test/conftest.py

```diff
@@ -170,16 +170,74 @@ def git_urls():
             'name': 'repo',
             'owner': 'owner',
         },
+        'https://example.com/owner/repo': {
+            'pathname': '/owner/repo',
+            'protocols': ['https'],
+            'protocol': 'https',
+            'href': 'https://example.com/owner/repo',
+            'resource': 'example.com',
+            'user': None,
+            'port': None,
+            'name': 'repo',
+            'owner':'owner',
+        },
+        'example.com:repo.git':{
+            'pathname': 'repo.git',
+            'protocols': [],
+            'protocol': 'ssh',
+            'href': 'example.com:repo.git',
+            'resource': 'example.com',
+            'user': None,
+            'port': None,
+            'name': 'repo',
+            'owner': None,
+        },
+        'git+https://example.com/owner/repo.git':{
+            'pathname': '/owner/repo.git',
+            'protocols': ['git', 'https'],
+            'protocol': 'https',
+            'href': 'git+https://example.com/owner/repo.git',
+            'resource': 'example.com',
+            'user': None,
+            'port': None,
+            'name': 'repo',
+            'owner': 'owner',
+        },
+        'https://example.com/owner':{
+            'pathname': '/owner',
+            'protocols': ['https'],
+            'protocol': 'https',
+            'href': 'https://example.com/owner',
+            'resource': 'example.com',
+            'user': None,
+            'port': None,
+            'name': None,
+            'owner': 'owner',
+        },
+        'https://example.in/owner':{
+            'pathname': '/owner',
+            'protocols': ['https'],
+            'protocol': 'https',
+            'href': 'https://example.in/owner',
+            'resource': 'example.in',
+            'user': None,
+            'port': None,
+            'name': None,
```

Comment:
this is wrong. the repo *must* have a name.  the https git access will expand this to be `owner.git` .  A repo doesnt need to have an owner; owners are a conceptual addition provided by git hosters.

Reply:
@jayvdb, I was thinking about this url https://github.com/coala/

Reply:
`https://github.com/jayvdb` is an invalid github remote url.
`https://gitlab.com/jayvdb` is an invalid gitlab remote url.
etc.

that is because we *know* those sites host repos under user/org/group prefix.

but the git world is not only github & gitlab.  While a library may give special treatment to very common cases, it should not do so at the expense of other git hosting.

`https://example.com/jayvdb` is assumed to be a git remote url for a repo called `jayvdb.git` hosted by `example.com`.  that is the only possible way to parse it, and not parsing it that way means that valid URL would be rejected, which would be a bug.